src/codechandler/AVCCodecHandler.cpp

```diff
@@ -0,0 +1,203 @@
+#include "AVCCodecHandler.h"
+
+AVCCodecHandler::AVCCodecHandler(AP4_SampleDescription* sd)
+  : CodecHandler(sd), m_countPictureSetIds(0), m_needSliceInfo(false)
+{
+  unsigned int width(0), height(0);
+  if (AP4_VideoSampleDescription* video_sample_description =
+          AP4_DYNAMIC_CAST(AP4_VideoSampleDescription, m_sampleDescription))
+  {
+    width = video_sample_description->GetWidth();
+    height = video_sample_description->GetHeight();
+  }
+  if (AP4_AvcSampleDescription* avc =
+          AP4_DYNAMIC_CAST(AP4_AvcSampleDescription, m_sampleDescription))
+  {
+    m_extraData.SetData(avc->GetRawBytes().GetData(), avc->GetRawBytes().GetDataSize());
+    m_countPictureSetIds = avc->GetPictureParameters().ItemCount();
+    m_naluLengthSize = avc->GetNaluLengthSize();
+    m_needSliceInfo = (m_countPictureSetIds > 1 || !width || !height);
+    switch (avc->GetProfile())
+    {
+      case AP4_AVC_PROFILE_BASELINE:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileBaseline;
+        break;
+      case AP4_AVC_PROFILE_MAIN:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileMain;
+        break;
+      case AP4_AVC_PROFILE_EXTENDED:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileExtended;
+        break;
+      case AP4_AVC_PROFILE_HIGH:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileHigh;
+        break;
+      case AP4_AVC_PROFILE_HIGH_10:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileHigh10;
+        break;
+      case AP4_AVC_PROFILE_HIGH_422:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileHigh422;
+        break;
+      case AP4_AVC_PROFILE_HIGH_444:
+        m_codecProfile = STREAMCODEC_PROFILE::H264CodecProfileHigh444Predictive;
+        break;
+      default:
+        m_codecProfile = STREAMCODEC_PROFILE::CodecProfileUnknown;
+        break;
+    }
+  }
+}
+
+bool AVCCodecHandler::ExtraDataToAnnexB()
+{
+  if (AP4_AvcSampleDescription* avc =
+          AP4_DYNAMIC_CAST(AP4_AvcSampleDescription, m_sampleDescription))
+  {
+    //calculate the size for annexb
+    size_t sz(0);
```

Comment:
```suggestion
    AP4_Size size{0};
```
better use the same type of the getter's/setter's